L1Trigger/TrackFindingTracklet/interface/MemoryBase.h

```diff
@@ -0,0 +1,51 @@
+#ifndef L1Trigger_TrackFindingTracklet_interface_MemoryBase_h
+#define L1Trigger_TrackFindingTracklet_interface_MemoryBase_h
+
+#include "L1Trigger/TrackFindingTracklet/interface/Settings.h"
+
+#include <fstream>
+#include <sstream>
+#include <cassert>
+#include <bitset>
+
+namespace Trklet {
+
+  class MemoryBase {
+  public:
+    MemoryBase(std::string name, const Settings* const settings, unsigned int iSector);
+
+    virtual ~MemoryBase() {}
+
+    std::string getName() const { return name_; }
+    std::string getLastPartOfName() const { return name_.substr(name_.find_last_of('_') + 1); }
+
+    virtual void clean() = 0;
+
+    //method sets the layer and disk based on the name. pos is the position in the memory name where the layer or disk is specified
+    void initLayerDisk(unsigned int pos, int& layer, int& disk);
+
+    unsigned int initLayerDisk(unsigned int pos);
+
+    // Based on memory name check if this memory is used for special seeding:
+    // overlap is layer-disk seeding
+    // extra is the L2L3 seeding
+    // extended is the seeding for displaced tracks
+    void initSpecialSeeding(unsigned int pos, bool& overlap, bool& extra, bool& extended);
+
+    //Used for a hack below due to MAC OS case sensitiviy problem for files
+    void findAndReplaceAll(std::string& data, std::string toSearch, std::string replaceStr);
+
+    void openFile(bool first, std::string filebase);
+
+  protected:
+    std::string name_;
+    unsigned int iSector_;
+
+    std::ofstream out_;
+    int bx_;
+    int event_;
+
+    const Settings* const settings_;
```

Comment:
can you please explain why this is better? for now we are keeping this as is since marked as "suggestion" but happy to note it down for future improvements.

Reply:
A pointer can be null, while a reference cannot be "null".
The downside of a reference is that you cannot reassign to point to a different object - but since you are using a `const` pointer (not just a pointer to `const`), you don't plan to reassign anyway.

Reply:
OK, I have now changed throughout code to pass Settings as const& instead of const pointer.